src/sinks/util/zstd.rs

```diff
@@ -0,0 +1,76 @@
+use std::{fmt::Display, io};
+
+use super::buffer::compression::CompressionLevel;
+
+#[derive(Debug)]
+pub struct ZstdCompressionLevel(i32);
+
+impl From<CompressionLevel> for ZstdCompressionLevel {
+    fn from(value: CompressionLevel) -> Self {
+        let val: i32 = match value {
+            CompressionLevel::None => 0,
+            CompressionLevel::Default => zstd::DEFAULT_COMPRESSION_LEVEL,
+            CompressionLevel::Best => 21,
+            CompressionLevel::Fast => 1,
+            CompressionLevel::Val(v) => {
+                if v < 1 {
+                    1
+                } else if v > 21 {
+                    21
+                } else {
+                    v as i32
+                }
+            }
```

Comment:
Could this use `clamp`?
```suggestion
            CompressionLevel::Val(v) => v.clamp(1, 21) as i32,
```